common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -228,7 +228,7 @@ private synchronized boolean connect() throws IOException {
         String localHostName = NetworkAddressUtils.getLocalHostName(mTachyonConf);
         LOG.info("Trying to get local worker host : " + localHostName);
         workerNetAddress = mMasterClient.user_getWorker(false, localHostName);
-        mIsLocal = true;
+        mIsLocal = NetworkAddressUtils.getFqdnHost(workerNetAddress).equals(localHostName);
```

Comment:
Maybe this mIsLocal flag should be set after the worker is connected? Does the call on line 242 guarantee a remote worker?

Also, is this the correct way for matching two addresses? The code in the master looks at: getHostName(), getHostAddress(), and getCanonicalHostName().